packages/components/src/test/tree/controlled/UsePresentationTreeState.test.ts

```diff
@@ -280,15 +279,22 @@ describe("usePresentationTreeState", () => {
       });
     });
 
-    it("creates a new nodeLoader when briefcase `onChangesApplied` event is raised", async () => {
+    it("creates a new nodeLoader when briefcase `onChangesPulled` event is raised", async () => {
       isBriefcaseConnectionStub.returns(true);
       const { result } = renderHook((props: UsePresentationTreeStateProps) => usePresentationTreeState(props), { initialProps });
       const oldNodeLoader = await waitForState(result);
 
-      expect(addListenerStub).to.be.calledOnce;
-      const listener = addListenerStub.getCall(0).args[0];
+      onChangesPulledEvent.raiseEvent();
 
-      listener();
+      await waitFor(() => expect(result.current?.nodeLoader).to.not.eq(oldNodeLoader));
+    });
+
+    it("creates a new nodeLoader when briefcase `onChangesPushed` event is raised", async () => {
```

Comment:
why do we need to reload on push?

Reply:
When pushing, the changes are pulled, merged and applied. I originally reloaded on the `onChangesApplied` event, however that does not always work as the connection can be closed right after the event is raised, which leads to the same issue. The `onChangesPulled` and `onChangesPushed` events are raised after the changes are applied and the connection is reopened for the last time, so the reload occurs after the operation is fully completed.